File: mobile/ios/OmenIOS/OmenIOS/DesignSystem/OmenPlatformBadge.swift
import SwiftUI

/// Registry §3.1 PlatformBadge: provider identity label. The literal name always renders —
/// color is a redundant signal, never the only signal (registry §4; facts-of-record #7).
///
/// Scope note (2026-07-21): registry §2.3 also names `-chip` legibility overrides and
/// `on-platform-*` foreground tokens that are not yet defined in `OmenColor`. This primitive
/// uses the same tinted-surface recipe as `OmenBadge` so it passes AA on both surface1 and
/// bg without depending on tokens that don't exist yet. Swap to fill-on-platform once the
/// registry token expansion lands.
enum OmenPlatform { case sleeper, yahoo, espn }

struct OmenPlatformBadge: View {
    let platform: OmenPlatform

    private var platformColor: Color {
        switch platform {
        case .sleeper: return OmenColor.Data.platformSleeper
        case .yahoo:   return OmenColor.Data.platformYahoo
        case .espn:    return OmenColor.Data.platformEspn
        }
    }

    private var label: String {
        switch platform {
        case .sleeper: return "Sleeper"
        case .yahoo:   return "Yahoo"
        case .espn:    return "ESPN"
        }
    }

    var body: some View {
        Text(label)
            .omenTextStyle(OmenTypography.chip)
            .foregroundStyle(platformColor)
            .padding(.horizontal, OmenSpacing.step8)
            .padding(.vertical, OmenSpacing.step4)
            .background(platformColor.opacity(0.15))
            .clipShape(Capsule())
    }
}

File: mobile/android/core/designsystem/src/main/kotlin/com/slopssaloon/omen/core/designsystem/component/OmenPlatformBadge.kt
package com.slopssaloon.omen.core.designsystem.component

import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.slopssaloon.omen.core.designsystem.theme.OmenTheme

/**
 * Registry §3.1 PlatformBadge: provider identity label using the invariant `platform-*` tokens
 * from registry §2.3. The literal provider name always renders — color is a redundant signal,
 * never the only signal (registry §4 accessibility contract; facts-of-record #7 rule that
 * data source must be labeled, not implied).
 *
 * Scope note (2026-07-21): registry §2.3 also names `-chip` legibility overrides and
 * `on-platform-*` foreground tokens, but those are not yet defined in [OmenColor.kt]. This
 * primitive uses the same tinted-surface recipe as [OmenBadge] (platform color at 15% alpha
 * fill, platform color as text) so it passes AA on both surface1 and bg without depending on
 * tokens that don't exist yet. Swap to fill-on-platform once the registry token expansion
 * lands.
 */
enum class OmenPlatform { Sleeper, Yahoo, Espn }

@Composable
fun OmenPlatformBadge(
    platform: OmenPlatform,
    modifier: Modifier = Modifier,
) {
    val platformColor: Color = when (platform) {
        OmenPlatform.Sleeper -> OmenTheme.color.data.platformSleeper
        OmenPlatform.Yahoo -> OmenTheme.color.data.platformYahoo
        OmenPlatform.Espn -> OmenTheme.color.data.platformEspn
    }
    val label = when (platform) {
        OmenPlatform.Sleeper -> "Sleeper"
        OmenPlatform.Yahoo -> "Yahoo"
        OmenPlatform.Espn -> "ESPN"
    }

    Surface(
        modifier = modifier,
        shape = RoundedCornerShape(999.dp),
        color = platformColor.copy(alpha = 0.15f),
        contentColor = platformColor,
    ) {
        Text(
            text = label,
            modifier = Modifier.padding(horizontal = OmenTheme.spacing.step8, vertical = OmenTheme.spacing.step4),
            style = OmenTheme.typography.chip.toTextStyle(),
        )
    }
}
